src/components/App/App.jsx

```diff
@@ -9,34 +10,45 @@ import Footer from '../Footer/Footer';
 import WithWrapper from '../WithWrapper/WithWrapper';
 import ModalsWrapper from '../ModalsWrapper/ModalsWrapper';
 import { BrowserRouter, Switch, Route } from 'react-router-dom';
+import NotFoundPage from 'components/NotFoundPage/NotFoundPage';
 
 const MainContentWithWrapper = WithWrapper(MainContent);
 
-const App = ({ closeModal, isModalShown, setMoviesThunk, setIsLoading }) => {
+const App = ({ closeModal, isModalShown, setMoviesByTerms }) => {
   const escapeHandler = e => e.which === 27 && isModalShown ? closeModal() : void 0;
-
-  useEffect(async () => {
-    setIsLoading(true);
-    await setMoviesThunk();
-    setIsLoading(false);
+  
+  useEffect(() => {
+    const url = new URL(window.location);
+    const searchParams = Object.fromEntries(url.searchParams);
+    setMoviesByTerms(searchParams);
   }, []);
 
+  const pageBottom = (
+    <>
+      <MainContentWithWrapper />
+      <Footer />
+      <ModalsWrapper />
+    </>
+  );
   return (
     <div onKeyUp={escapeHandler} className={styles.App}>
       <BrowserRouter>
         <Switch>
-          <Route exact path='/' component={Header} />
-          <Route path='/movie/:id' component={MovieHeader} />
+          <Route exact path='/'>
+            <Header />
+            { pageBottom }
```

Comment:
how about moving this to separate component (page) so file with routes won't know anything about the inner implementation (same for lines:42-43). (Good place to benefit from composition)

Thus you will have clear and concise file with routes. 
And if something changes in Pageimplementation -> component with routes won't know about it (because it's not its responsibility to be aware of inner implementation. **ONLY** thing it should care about is have route and show appropriate component)